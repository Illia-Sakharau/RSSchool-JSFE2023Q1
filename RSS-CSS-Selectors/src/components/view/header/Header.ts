import './header.scss';
import htmlToElement from '../../../utils/htmlToElement';
import createElement from '../../../utils/createElement';
import logo from '../../../assets/logo.svg';
import { CURRENT_LEVEL } from '../../../data/constants';

export default class Header {
  public draw(): HTMLElement {
    const template: string = `<header>
        <div class="logo">
          ${logo}
          <h1>Floral CSS</h1>
        </div>
        <div class="level">
          <h2>Level ${CURRENT_LEVEL.getLevel() + 1}:</h2>
          <span>${CURRENT_LEVEL.getTitle()}</span>
        </div>
      </header>
    `;
    const headerEl: HTMLElement = htmlToElement(template);
    const menuEl: HTMLElement = createElement({ tag: 'div', classes: ['menu'] });
    menuEl.appendChild(createElement({ tag: 'div', classes: ['line'] }));

    // open/close aside meny
    menuEl.addEventListener('click', () => {
      const event = new Event('toggleMenu');
      document.dispatchEvent(event);
    });
    document.addEventListener('toggleMenu', () => {
      menuEl.classList.toggle('menu_close');
    });
    document.addEventListener('closeMenu', () => {
      menuEl.classList.remove('menu_close');
    });

    headerEl.appendChild(menuEl);

    return headerEl;
  }
}
