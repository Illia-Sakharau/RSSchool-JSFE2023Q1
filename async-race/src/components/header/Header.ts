import './header.scss';
import htmlToElement from '../../utils/htmlToElement';
import createElement from '../../utils/createElement';
import logoImg from '../../assets/logo.svg';
import gitImg from '../../assets/github.svg';
import schoolLogoImg from '../../assets/rsslogo.svg';
import { ROUTES } from '../../data/constants';
// import { router } from '../../router/router';

function createLogo(): HTMLElement {
  const template: string = `<div class="logo">
      ${logoImg}
      <h1>async-race</h1>
    </div>
  `;
  const logoEl: HTMLElement = htmlToElement(template);

  return logoEl;
}

function createMenu(activeLinkText: string): HTMLElement {
  const menuEl: HTMLElement = createElement({ tag: 'nav', classes: ['menu'] });
  ROUTES.forEach((route) => {
    const classes = activeLinkText === route.text ? ['menu__link', 'menu__link_active'] : ['menu__link'];
    const linkEl: HTMLElement = createElement({ tag: 'a', classes, content: route.text });
    linkEl.setAttribute('href', route.path);
    menuEl.append(linkEl);
  });

  // menuEl.addEventListener('click', (event) => {
  //   const target = event.target;
  //   if (target instanceof HTMLAnchorElement && target.className === 'menu__link') {
  //     event.preventDefault();
  //     history.pushState(null, '', target.href);
  //     router();
  //   }
  // });

  return menuEl;
}

function createInfo(): HTMLElement {
  const template: string = `<div class="info">
      <a href="https://rs.school/js/" target="_blank" class="info__school">
        ${schoolLogoImg}
      </a>
      <div class="info__wrapper">
        <a href="https://github.com/Illia-Sakharau" target="_blank" class="info__author">
          ${gitImg}
          <span>Illia Sakharau</span>
        </a>        
        <div class="info__copyright">
          Â© 2023
        </div>
      </div>
    </div>
    `;
  const infoEl: HTMLElement = htmlToElement(template);

  return infoEl;
}

export default function createHeader(activeLinkText: string): HTMLElement {
  const headerEl: HTMLElement = createElement({ tag: 'header', classes: ['header'] });
  const logoEl: HTMLElement = createLogo();
  const menuEl: HTMLElement = createMenu(activeLinkText);
  const infoEl: HTMLElement = createInfo();

  logoEl.classList.add('header__logo');
  menuEl.classList.add('header__menu');
  infoEl.classList.add('header__info');

  headerEl.append(logoEl, menuEl, infoEl);

  return headerEl;
}
