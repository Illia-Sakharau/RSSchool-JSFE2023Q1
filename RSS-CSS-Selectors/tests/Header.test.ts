import Header from '../src/components/view/header/Header';

jest.mock('../src/data/constants', () => {
  return {
    CURRENT_LEVEL: {
      getLevel: () => 0,
      getTitle: () => 'Test title',
    },
  };
});
jest.mock('../src/components/view/header/header.scss', () => ({}));
jest.mock('../src/assets/logo.svg', () => 'logo.svg');
const header = new Header();

afterAll(() => {
  jest.restoreAllMocks();
});

describe('Test drow header', () => {
  test('should return correct answer', () => {
    expect(header.draw()).toBeDefined();
  });
  test('should return header', () => {
    expect(header.draw()).toMatchSnapshot();
  });
});
