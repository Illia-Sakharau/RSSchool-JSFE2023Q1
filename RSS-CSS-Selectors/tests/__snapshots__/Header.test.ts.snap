// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test drow header should return header 1`] = `
<header>
  
        
  <div
    class="logo"
  >
    
          logo.svg
          
    <h1>
      Floral CSS
    </h1>
    
        
  </div>
  
        
  <div
    class="level"
  >
    
          
    <h2>
      Level 1:
    </h2>
    
          
    <span>
      Test title
    </span>
    
        
  </div>
  
      
  <div
    class="menu"
  >
    <div
      class="line"
    />
  </div>
</header>
`;
